sdk/servicebus/service-bus/src/serviceBusAtomManagementClient.ts

```diff
@@ -0,0 +1,593 @@
+ *
+ */
+export class ServiceBusAtomManagementClient extends ServiceClient {
+  private endpoint: string;
+
+  private queueResourceSerializer: AtomXmlSerializer;
+  private topicResourceSerializer: AtomXmlSerializer;
+  private subscriptionResourceSerializer: AtomXmlSerializer;
+  private ruleResourceSerializer: AtomXmlSerializer;
+
+  /**
+   * Initializes a new instance of the ServiceBusManagementClient class.
+   * @param credentials Credentials needed for the client to connect to Azure.
+   * @param options The parameter options
+   */
+  constructor(connectionString: string, proxySettings?: ProxySettings) {
+    const connectionStringObj = ServiceBusAtomManagementClient.parseConnectionString(
+      connectionString
+    );
+    const credentials = new SasServiceClientCredentials(
+      connectionStringObj.SharedAccessKeyName,
+      connectionStringObj.SharedAccessKey
+    );
+
+    const requestPolicyFactories: RequestPolicyFactory[] = [];
+    requestPolicyFactories.push(userAgentPolicy());
+    requestPolicyFactories.push(logPolicy(httpAtomXml));
+    requestPolicyFactories.push(atomSerializationPolicy());
+    requestPolicyFactories.push(signingPolicy(credentials));
+
+    if (proxySettings) {
+      requestPolicyFactories.push(proxyPolicy(proxySettings));
+    }
+    const serviceClientOptions: ServiceClientOptions = {
+      requestPolicyFactories: requestPolicyFactories
+    };
+
+    super(credentials, serviceClientOptions);
+    this.queueResourceSerializer = new QueueResourceSerializer();
+    this.topicResourceSerializer = new TopicResourceSerializer();
+    this.subscriptionResourceSerializer = new SubscriptionResourceSerializer();
+    this.ruleResourceSerializer = new RuleResourceSerializer();
+    this.endpoint = (connectionString.match("Endpoint=sb://(.*)/;") || "")[1];
+  }
+
+  /**
+   * Creates a queue.
+   * @param queuePath Name of the queue to use
+   * @param queueOptions
+   */
+  async createQueue(queuePath: string, queueOptions: QueueOptions): Promise<QueueResult> {
+    const response: HttpOperationResponse = await this._putResource(
+      queuePath,
+      queueOptions,
+      false,
+      this.queueResourceSerializer
+    );
+    return this.buildResponse(response);
+  }
+
+  /**
+   * Gets a queue.
+   * @param queuePath Name of the queue
+   */
+  async getQueue(queuePath: string): Promise<QueueResult> {
+    const response: HttpOperationResponse = await this._getResource(
+      queuePath,
+      this.queueResourceSerializer
+    );
+    return this.buildResponse(response);
+  }
+
+  /**
+   * Lists existing queues.
+   * @param listRequestOptions
+   */
+  async listQueues(listRequestOptions?: ListRequestOptions): Promise<ListQueuesResult> {
+    const response: HttpOperationResponse = await this._listResources(
+      "$Resources/Queues",
+      listRequestOptions,
+      this.queueResourceSerializer
+    );
+    return this.buildResponse(response);
+  }
+
+  /**
+   * Updates a queue.
```

Comment:
```suggestion
   * Updates properties on the queue by the given name based on the given options
```

Please use the above format for docs for all the other update operations as well